### PR TITLE
Tuned regex for normal escapes

### PR DESCRIPTION
**Pull request by DavyLandman**
I've added support for `\xHH` escapes for ascii chars, and also enabled `\r\n\t` and friends.

### CODE REVIEW

File: vscode-extension/syntaxes/rascal.tmLanguage.json [hunk 1]
```diff
@@ -19,16 +19,16 @@
...
       "name": "prod(lex(\"delimiters\"),[alt({lit(\"bottom-up-break\"),lit(\")\"),lit(\"≫\"),lit(\"\\n\"),lit(\"!:=\"),lit(\"\\'\"),lit(\"!=\"),lit(\"\\>=\"),lit(\"://\"),lit(\"non-assoc\"),lit(\"&=\"),lit(\"\\<-\"),lit(\"*=\"),lit(\"+=\"),lit(\"top-down-break\"),lit(\",\"),lit(\"...\"),lit(\"/=\"),lit(\"⇨\"),lit(\"(\"),lit(\"*/\"),lit(\"%\"),lit(\"!\\<\\<\"),lit(\"=\\>\"),lit(\"!\\>\\>\"),lit(\"||\"),lit(\"\\>\\>\"),lit(\"::\"),lit(\"⚠\"),lit(\"&&\"),lit(\"@\"),lit(\":=\"),lit(\"#\"),lit(\"\\<==\\>\"),lit(\"\\\"\"),lit(\"\\<\\<=\"),lit(\"}\"),lit(\"?=\"),lit(\"\\<:\"),lit(\"==\\>\"),lit(\"^\"),lit(\"/*\"),lit(\";\"),lit(\"{\"),lit(\"-=\"),lit(\"$T\")})],{})",
-      "captures": []
+      "captures": {}
```
Comment (DavyLandman) [inline vscode-extension/syntaxes/rascal.tmLanguage.json:24]: this is a side effect of latest rascal version that changes how empty maps are encoded. (it used to be a list, but now its a empty object.

File: rascal-textmate-core/src/main/rascal/lang/oniguruma/Conversion.rsc [hunk 2]
```diff
@@ -184,5 +188,24 @@ private list[str] hex
...
+
+private map[int, str] escapes = (
+    0x09: "\\t",
+    0x0A: "\\n",
+    0x0D: "\\r",
+    0x20: "\\x20" // spaces look a bit strange in a regex, although they are valid, people tend to read over them as layout
+) + ( c : "\\<stringChar(c)>" | c <- [0x21..0x7F], c notin printable); // regular ascii characters that might have special meaning in a regex
+
+
+private map[int, str] addFallBack(map[int, str] defined)
```
Comment (sungshik) [inline rascal-textmate-core/src/main/rascal/lang/oniguruma/Conversion.rsc:205]:
```suggestion
private map[int, str] addFallback(map[int, str] defined)
```

File: rascal-textmate-core/src/main/rascal/lang/oniguruma/Conversion.rsc [hunk 3]
```diff
@@ -184,5 +188,24 @@ private list[str] hex
...
+    = ( char : "\\x<right(toHex(char),2, "0")>" | char <- [0..256], char notin defined)
+    + defined
+    ;
+
+private map[int, str] preEncoded
+    = addFallBack(escapes + ( c : stringChar(c) | c <- printable));
```
Comment (sungshik) [inline rascal-textmate-core/src/main/rascal/lang/oniguruma/Conversion.rsc:211]:
```suggestion
    = addFallback(escapes + ( c : stringChar(c) | c <- printable));
```

File: rascal-textmate-core/src/main/rascal/lang/oniguruma/Conversion.rsc [hunk 4]
```diff
@@ -170,10 +170,14 @@ str encode(list[int] chars, bool withBounds = false)
...
     : intercalate("", [encode(i) | i <- chars]);
 
-str encode(int char)
-    = char in alnum 
-    ? stringChar(char)
-    : "\\x{<toHex(char)>}";
+str encode(int char) 
+    = char in preEncoded
+    ? preEncoded[char]
+    : "\\x{<toHex(char)>}"
```
Comment (sungshik) [inline rascal-textmate-core/src/main/rascal/lang/oniguruma/Conversion.rsc:176]:
```suggestion
    = preEncoded[char]
    ? "\\x{<toHex(char)>}"
```

(If that's indeed equivalent)

**Review by sungshik — APPROVED** [review]
Thanks, LGTM, just a few nits

File: rascal-textmate-core/src/main/rascal/lang/oniguruma/Conversion.rsc (hunk 4 above)
Comment (DavyLandman) [reply rascal-textmate-core/src/main/rascal/lang/oniguruma/Conversion.rsc:176]: Yes, I actually did it for performance reasons, but I shouldn't, unless it's slow.

**Pull request merged**